internal/updater/updater.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 The Operator-SDK Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package updater
```

Comment:
IIRC, this was originally from the helm-operators-plugin repository, so we'd probably want to start there? cc @joelanford

Reply:
It would be nice to break it out in it's own package. Each updater is specific to a particular type (for inspecting that particular types' status). It seems like a good generic problem that I've been trying to solve. We could potentially have a factory that can produce an Updater for any given type that has a Status field.